qcodes/dataset/measurements.py

```diff
@@ -296,7 +398,8 @@ def __init__(
         self.parameters = parameters
         # here we use 5 s as a sane default, but that value should perhaps
         # be read from some config file
-        self.write_period = write_period if write_period is not None else 5
+        self.write_period = float(write_period) \
+            if write_period is not None else 5.0
```

Comment:
why is this int/float business needed for write_period? also, if it is needed, just keep it in the `write_period.setter` only, there is no reason for other methods to care about this.

Reply:
The existing logic was wrong and only worked because nothing was actually checked by mypy. You cannot use the Number ABC as a type. So I opted for making it float internally. The alternative is just make everything a union[float, int] I don't really care either which way.

Reply:
ok. now, the solution is fine, it's just that i'd rather have `float(..)` calls in one place.